Sources/SwiftParser/Lexer.swift

```diff
@@ -1333,20 +1338,20 @@ extension Lexer.Cursor {
       // 0b[01][01_]*
       _ = self.advance()
       if !self.isAtEndOfFile, self.peek() != UInt8(ascii: "0") && self.peek() != UInt8(ascii: "1") {
-//        return expected_int_digit(self, .binary)
+        diagnosticHandler?(TokStart.distance(to: self), InvalidDigitInIntegerliteral(kind: .binary(Unicode.Scalar(self.peek()))))
         self.advance(while: { $0.isValidIdentifierContinuationCodePoint })
-        return (.unknown, [])
+        return (.integerLiteral, [ .isErroneous ])
       }
 
       self.advance(while: {
         $0 == Unicode.Scalar("0") || $0 == Unicode.Scalar("1") || $0 == Unicode.Scalar("_")
       })
 
-//      let tmp = self
+      let tmp = self
```

Comment:
Why do we need `tmp`? Can’t we use `self` below? Same in all the other cases.

Reply:
That would be an off-by-one error

Reply:
Ah, I missed the `advance` below. Makes sense 👍